tensorflow/python/keras/utils/generic_utils_test.py

```diff
@@ -74,12 +73,116 @@ class CustomClass(object):
 class SerializeKerasObjectTest(test.TestCase):
 
   def test_serialize_none(self):
-    serialized = keras.utils.generic_utils.serialize_keras_object(None)
+    serialized = generic_utils.serialize_keras_object(None)
     self.assertEqual(serialized, None)
-    deserialized = keras.utils.generic_utils.deserialize_keras_object(
-        serialized)
+    deserialized = generic_utils.deserialize_keras_object(serialized)
     self.assertEqual(deserialized, None)
 
 
+class FuncDumpAndLoadTest(test.TestCase, parameterized.TestCase):
+  @parameterized.parameters(['simple_function', 'closured_function'])
+  def test_func_dump_and_load(self, test_function_type):
+    if test_function_type == 'simple_function':
+      def test_func():
+        return r'\u'
+
+    elif test_function_type == 'closured_function':
+      def get_test_func():
+        x = r'\u'
+
+        def test_func():
+          return x
+        return test_func
+      test_func = get_test_func()
+    else:
+      raise Exception('Unknown test case for test_func_dump_and_load')
+
+    serialized = generic_utils.func_dump(test_func)
+    deserialized = generic_utils.func_load(serialized)
+    self.assertEqual(deserialized.__code__, test_func.__code__)
+    self.assertEqual(deserialized.__closure__, test_func.__closure__)
+    self.assertEqual(deserialized.__defaults__, test_func.__defaults__)
+
+  def test_func_dump_and_load_closure(self):
+    y = 0
+    test_func = lambda x: x + y
+    serialized, _, closure = generic_utils.func_dump(test_func)
+    deserialized = generic_utils.func_load(serialized, closure=closure)
+    self.assertEqual(deserialized.__code__, test_func.__code__)
+    self.assertEqual(deserialized.__closure__, test_func.__closure__)
+    self.assertEqual(deserialized.__defaults__, test_func.__defaults__)
+
+
+  @parameterized.parameters(
+      [keras.activations.softmax, np.argmax, lambda x: x**2, lambda x: x])
+  def test_func_dump_and_load_backwards_compat(self, test_func):
+    serialized = marshal.dumps(test_func.__code__).decode(
+        'raw_unicode_escape')
+
+    deserialized = generic_utils.func_load(
+        serialized, defaults=test_func.__defaults__)
+    self.assertEqual(deserialized.__code__, test_func.__code__)
+    self.assertEqual(deserialized.__closure__, test_func.__closure__)
+    self.assertEqual(deserialized.__defaults__, test_func.__defaults__)
+
+
+class ProgbarTest(test.TestCase):
+
+  def test_progbar(self):
+    values_s = [None,
+                [['key1', 1], ['key2', 1e-4]],
+                [['key3', 1], ['key2', 1e-4]]]
+
+    for target in (len(values_s) - 1, None):
+      for verbose in (0, 1, 2):
+        bar = generic_utils.Progbar(
```

Comment:
For a progbar test, you should catch the stdout and check that the display matches what you expect in the different verbosity modes.